docs/reference/search/rrf.asciidoc

```diff
@@ -1,6 +1,423 @@
 [[rrf]]
 === Reciprocal rank fusion
 
-Reciprocal Rank Fusion (RRF) is a simple method to combine document result sets
-from multiple queries where the queries' document scores may be unrelated.
+NOTE: Reciprocal rank fusion requires a Platinum subscription or higher.
 
+IMPORTANT: Reciprocal rank fusion is in technical preview. Features in technical
+preview may be changed or removed in a future release. Elastic will apply best
+effort to fix any issues, but features in technical preview are not subject to
+the support SLA of official GA features.
+
+Reciprocal rank fusion (RRF) is a method for combining results from
+multiple queries with different relevance indicators into a single
+result set where there is no requirement the relevance indicators for
+each query are related to each other in order to achieve high-quality results.
+
+RRF uses the following formula to determine the score for ranking each document:
+
+[source,python]
+----
+score = 0.0
+for q in queries:
+    if d in result(q):
+        score += 1.0 / ( k + rank( result(q), d ) )
+return score
+
+# where
+# k is a ranking constant
+# q is a query in the set of queries
+# d is a document in the result set of q
+# result(q) is the result set of q
+# rank( result(q), d ) is d's rank within the result(q) starting from 1
+----
+// NOTCONSOLE
+
+[[rrf-api]]
+==== Reciprocal rank fusion API
+
+RRF may be used as part of a <<search-search, search>> to combine and rank
+documents using either a single standard query and 1-to-many knn searches or no standard
+query and 2-to-many knn searches. The `rrf` parameter is an optional object defined as part
+of a search request's <<request-body-rank, rank parameter>>.
+
+The `rrf` object contains the following parameters:
+
+`rank_constant`::
+(Optional, integer) This value determines how much influence documents in individual
+result sets per query have over the final ranked result set. A higher value indicates
+that lower ranked documents have more influence. This value must be greater than or
+equal to `1`. Defaults to `60`.
+
+`window_size`::
+(Optional, integer) This value determines the size of the individual results sets per
+query. A higher value will improve result relevance at the cost of performance. The final
+ranked result set is pruned down to the search request's <<search-api-query-params-q, size>.
+`window_size` must be greater than or equal to `size` and greater than or equal to `1`.
+Defaults to `100`.
+
+An example request using RRF:
+
+[source,console]
+----
+GET rrf-index/_search
+{
+    "query": {                       <1>
+        "term": {
+            "rrf_text": "shoes"
```

Comment:
The name of the field _might_ imply that you need a special type of field to use rrf on it for standard searches? I'm guessing that's not true, but maybe leave "rrf" out of the field name to avoid confusion? (treat this suggestion as completely optional)

Reply:
Good point! I changed removed the `rrf_` from all of these and renamed the index to `example-index`.